app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -175,6 +194,71 @@ public boolean onPreferenceClick(Preference preference) {
         }
     }
 
+    /**
+     * Ask for location permission if the user agrees on attaching location with pictures
+     * and the app does not have the access to location
+     *
+     */
+    private void requestForLocationAccess() {
+        PermissionUtils.checkPermissionsAndPerformAction(getActivity(),
+            permission.ACCESS_FINE_LOCATION,
+            () -> onLocationPermissionGranted(getActivity()),
+            () -> {},
+            R.string.ask_to_turn_location_on,
```

Comment:
DItto on the title to be used here.

Speaking about dittos, could you look into avoiding duplication of these code in SettingsFragment and the ContributionController ? Duplication of the code is bad for maintenance. I believe it should be easy to make the code reusable. Let us know in case you have need any clarifications about the same.

Reply:
> Duplication of the code is bad for maintenance.

I usually do not write redundant code. I'm sorry but this is how the app is handling location related stuff currently. In fact, this needs to be standardised with utmost priority as this is leading to inconsistent behaviour at many places. [This](https://github.com/commons-app/apps-android-commons/pull/5205) is the link to my pull request where I'd started standardising it (it had the same permissions related code, and the same map behaviour was required in all the fragments but still duplication was high). 

Is it okay if I handle this separately?

Reply:
It's a pity that the app already has much duplicated instances. Let that not be a motivator to add more tech debt, though. Kindly look into avoiding the duplication in this case. Do share if you face any difficulties with it.

The standardization you mention seems like a good thing that could certainly be pursued later on. Feel free to open an issue to keep track of it 👍🏼